### DISCUSSION TITLE
The message in the Pulsar has spiked suddenly, causing increased query delays and making the service unavailable.

### DISCUSSION THREAD

**Discussion by Huangpingshan** [milvus-io/milvus#30291] [Q&A and General discussion]
Milvus version 2.2.9

View the Prometheus monitoring information for Pulsar,  there are several occurrences of sudden message spikes. Once the number of messages in Pulsar spikes to a certain threshold, the query latency increases, leading to service unavailability. Currently, we can only address this issue by clearing the messages in Pulsar and restarting the Pulsar components.

Previously, after we cleared and restarted Pulsar , the message backlog did not increase for the next few hours. However, starting from this evening, immediately after clearing and restarting Pulsar, the message began to accumulate rapidly, and the milvus service has essentially become nearly unusable.

![image](https://github.com/milvus-io/milvus/assets/84720705/408c0592-850b-4639-a7ff-767b26fd1eb4)

A peculiar observation is that, when examining the backlog of messages in each topic, it is found that only a few topics within the 'delta' category have a significant message backlog, while the 'dml' category contains very few messages.
![image](https://github.com/milvus-io/milvus/assets/84720705/d02d70d0-6c7e-4e8f-b288-a023cc0c3ae6)
![image](https://github.com/milvus-io/milvus/assets/84720705/44060036-d9e1-4556-8a7d-1546f5336c63)

How can we identify the source of this message backlog, and what measures should we take to address the issue now?  Whether it's the consumers or the producers that are causing the problem.

**Comment by yhmo** [comment]
We don't know your scenario and workflow. I guess the Pulsar becomes the bottleneck of your service because there are intensive/high-frequency insert/delete/upsert requests.
